examples/getting-started/skaffold.yaml

```diff
@@ -3,6 +3,10 @@ kind: Config
 build:
   artifacts:
   - imageName: gcr.io/k8s-skaffold/skaffold-example
+test:
```

Comment:
I can see pros and cons to listing the tests here. My first intent would have been to put them next to the artifacts. I think it makes the common case easier to write and avoid duplicating the image names. However, if one uses profiles, it's more complicated to write.

Reply:
I originally had them specified at the artifact level like you said, but changed it because I liked having an explicit "build/test/deploy" structure to the config. I agree though that even though the image name is required for the tests, doing it this way isn't as clear that the tests are artifact-specific, so it's a trade-off.

Reply:
Hmm...we could think about it in a spectrum way too: 
 - artifact level: container level tests and unittests. I think about these as "user specified compilation verification" - so quick, small and focused
 - build / test / deploy makes sense too - but probably would point at multi-container type integration tests using docker only?
 - integration tests/smoke tests could be in a verification phase after deploy

Reply:
Hmm, this is an interesting way to think about it. I'm definitely of the mind that container-level tests and application-level integration tests should be separate, so in that sense I like the idea of a "verification" phase after the deploy. The "test" phase should probably only operate at an artifact/single-container level.

By putting the test phase in between build and deploy, I wanted to make it more explicit that builds will always happen, but tests will only run if the builds pass, and similarly deploys will only happen if tests pass. Verification would similarly only happen if the deploys pass.

Reply:
Hi there 👋 

Just for your information, here are my "user" goal: _I want to be able to test a newly built image_
To do that in a artifact level, I think the main idea is to not deploy a bad container and get feedbacks from what I'm doing... So thank you for enabling ["Container Structure Test"](https://github.com/GoogleContainerTools/container-structure-test)

At this point, I don't have any preference between the `test` section inside the `build` section or outside. If it avoids to repeat the image name, that's cool 👍

Another important features:
- I think we don't want to run the test phase of an image that didn't move. (to improve my Developer eXperience)
- I think we want to use wildcards (like I do in the `deploy.kubectl.manifests` section)

I'm happy to help!

Reply:
Hey @jlandure, thanks for your feedback! This is exactly what I'm looking for :)

I agree, repeating the image name isn't great from a user standpoint, so I'll play around with specifying the tests at the artifact level (as opposed to through a separate `test` section) and see if it looks better.

Totally agree about not testing an image that didn't change, that should be covered by this PR. Only images that get rebuilt will trigger their specific tests.

Wildcards are definitely doable, I'll look into adding that.